numerai/cli/node/test.py

```diff
@@ -499,6 +507,133 @@ def azure_refresh_and_print_log(
     return monitoring_done, shown_log_row_key
 
 
+def monitor_gcp(node, config, verbose, log_type, trigger_id):
+    ## Write a function that uses the gcloud runv2 executions SDK to get executions that are currently running
+    monitor_start_time = datetime.now(timezone.utc) - timedelta(minutes=1)
+
+    gcp_key_path = get_gcp_keys()
+    os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = gcp_key_path
+    client = run_v2.ExecutionsClient()
+    
+    # Setup logging if necessary
+    previous_insert_id = "0"
+    logging_client = None
+    if verbose:
+        logging_client = logging_v2.Client()
+
+    time_lapse = datetime.now(timezone.utc) - monitor_start_time
+    monitoring_done = False
+
+    if verbose and log_type == LOG_TYPE_WEBHOOK:
+        print_gcp_webhook_logs(logging_client, config["job_id"])
+    
+    while time_lapse <= timedelta(minutes=15) and monitoring_done == False:
+        executions = get_gcp_job_executions(client, config["job_id"], trigger_id)
+        if len(executions) == 0:
+            click.secho(f"No job executions yet, still waiting...\r", fg="yellow")
+        else:
+            monitoring_done, status, previous_insert_id = check_gcp_execution_status(logging_client, config["job_id"], executions[0], verbose if log_type == LOG_TYPE_CLUSTER else False, previous_insert_id)
+            if monitoring_done or status == "unknown":
+                break
+        time.sleep(5 if verbose else 15)
+
+    if time_lapse >= timedelta(minutes=15):
+        click.secho(
+            f"Monitor timeout after 15 minutes, container run status cannot be determined. Recommended to check ran status directly on Google Cloud Console",
+            fg="red",
+        )
+        exit(1) 
+
+
+def get_gcp_job_executions(client, job_id, trigger_id):
+    # Initialize request argument(s)
+    request = run_v2.ListExecutionsRequest(
+        parent=job_id,
+    )
+
+    page_result = client.list_executions(request=request)
+    executions = []
+    # Handle the response
+    for response in page_result:
+        env = response.template.containers[0].env
+        for env_var in env:
+            if env_var.name == "TRIGGER_ID" and env_var.value == trigger_id:
+                executions.append(response)
+            elif trigger_id == None:
+                executions.append(response)
+
+    if trigger_id == None:
+        executions = [executions[0]]
+    return executions
+
+
+def check_gcp_execution_status(logging_client, job_id, execution, verbose, previous_insert_id):
```

Comment:
this whole function seems too verbose. It seems like this can be more readable and concise if we just use a dictionary mapping run_v2.types.Condition.State... to the output status we want.